utils/frame/sub-du/src/main.rs

```diff
@@ -0,0 +1,254 @@
+impl Module {
+	fn new(name: String) -> Self {
+		Self { name, ..Default::default() }
+	}
+}
+
+#[derive(Debug, Copy, Clone)]
+pub enum StorageItem {
+	Value(usize),
+	Map(usize, usize),
+}
+
+impl std::fmt::Display for StorageItem {
+	fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+		match self {
+			Self::Value(bytes) => write!(f, "Value({} bytes)", bytes.separated_string()),
+			Self::Map(bytes, count) => {
+				write!(f, "Map({} bytes, {} keys)", bytes.separated_string(), count)
+			}
+		}
+	}
+}
+
+impl Default for StorageItem {
+	fn default() -> Self {
+		Self::Value(0)
+	}
+}
+
+#[derive(Debug, Clone, Default)]
+struct Storage {
+	pub name: String,
+	pub size: usize,
+	pub item: StorageItem,
+}
+
+impl std::fmt::Display for Storage {
+	fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+		let item_style = Style::new().italic();
+		write!(f, "{} => {}", item_style.paint(self.name.clone()), self.item)
+	}
+}
+
+impl Storage {
+	fn new(name: String, item: StorageItem) -> Self {
+		let size = match item {
+			StorageItem::Value(s) => s,
+			StorageItem::Map(s, _) => s,
+		};
+		Self { name, item, size }
+	}
+}
+
+#[derive(Debug, StructOpt)]
+#[structopt(
+	name = "sub-du",
+	about = "a du-like tool that prints the map of storage usage of a substrate chain"
+)]
+struct Opt {
+	/// The block number at which the scrap should happen. Use only the hex value, no need for a
+	/// `0x` prefix.
+	#[structopt(long)]
+	at: Option<Hash>,
+
+	/// The node to connect to.
+	#[structopt(long, default_value = "ws://localhost:9944")]
+	uri: String,
+
+	/// If true, intermediate values will be printed.
+	#[structopt(long, short)]
+	progress: bool,
+
+	/// Weather to scrape all pairs or just the size of them.
+	///
+	/// If enabled, the command might take longer but then the number of keys in each map is also
+	/// scraped.
+	///
+	/// # Warning
+	///
+	/// This uses an unsafe RPC call and can only be used if the target node allows it.
+	#[structopt(long, short)]
+	scrape_pairs: bool,
+}
+
+#[async_std::main]
+async fn main() -> () {
+	env_logger::Builder::from_default_env().format_module_path(false).format_level(true).init();
+
+	let opt = Opt::from_args();
+
+	// connect to a node.
+	use jsonrpsee_ws_client::{WsClient, WsConfig};
+	let client = WsClient::new(&opt.uri, WsConfig::default()).await.unwrap();
+
+	let mut modules: Vec<Module> = vec![];
+
+	// potentially replace head with the given hash
+	let head = get_head(&client).await;
+	let at = opt.at.unwrap_or(head);
+	let runtime = sub_storage::get_runtime_version(&client, at).await;
+
+	println!("Scraping at block {:?} of {}({})", at, runtime.spec_name, runtime.spec_version,);
+
+	let raw_metadata = get_metadata(&client, at).await.0;
+	let prefixed_metadata = <RuntimeMetadataPrefixed as codec::Decode>::decode(&mut &*raw_metadata)
+		.expect("Runtime Metadata failed to decode");
+	let metadata = prefixed_metadata.1;
+
+	if let RuntimeMetadata::V12(inner) = metadata {
```

Comment:
support for metadata v13 and v14? :)

Reply:
yeah, that's the plan. I have not touched it much since reintroduction, on it now.

Reply:
@niklasad1 do we want to support both v13 and v14? I guess v12 Is probably not needed anymore.

Reply:
I guess I'll start with v14

Reply:
Sounds good to start with v14, gotcha I was just surprised when I spotted v12 :)

No sure might be nice to query on really old blocks to check how it grows over time not sure how use it will be.

Reply:
We should at first only support v14. no need to support the old versions initially. If you can do it, sure, but otherwise no need.